internal/provider/data_source_http.go

```diff
@@ -317,6 +317,11 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 		request.Header.Set(name, header)
 	}
 
+	hostHeader := request.Header.Get("Host")
+	if hostHeader != "" {
+		request.Host = hostHeader
+	}
+
```

Comment:
nit: This is slightly confusing to read, since it's not immediately obvious why the above `request.Header.Set` partially works, but needs to be extracted/moved later. I actually assumed that `request.Header.Set("Host", "val")` would do nothing after learning about the oddities of `net/http` 😆 

Perhaps a comment referencing the Go pkg docs for `request.Host` field being the correct final location for this header would help? Or maybe setting the `Host` header inside the for loop above? `If name == host, set on request directly, otherwise set in the headers.`

Reply:
That's fair. I had thought that `request.Header.Set()` was combining multiple keys to a single value, so that if someone specified multiple `Host` headers, we would replace the `request.Host` with concatenation of the request header values, but it seems that `request.Header.set()` just replaces the value in the loop, so I've moved the `request.Host` code to inside of the loop.